test/functional/sqlsrv/sqlsrv_ae_datetimes_as_strings.phpt

```diff
@@ -0,0 +1,595 @@
+
+}
+
+// The date used for the test will be Januray 31, 2002, or 2002/01/31.
+// This will sidestep issues involving the use of two digit years since 
+// SQL Server defaults to 19 as the first two digits.
+// Time is 23:59:29.049876
+// Note that smalldatetime rounds to the nearest minute, and that may cause 
+// this test to fail if it rolls over to the next day.
+// Incidentally, this datetime corresponds to a timestamp of 1012521599.
+$year = '2002';
+$month = '01';
+$month_name = 'January';
+$month_abbr = 'Jan';
+$day = '31';
+$hour = '23';
+$hour12 = '11';
+$meridian = 'PM';
+$minute = '59';
+$second = '29';
+$frac = '04';
+$frac2 = '9876';
+$tz_correction = '+08:00';
+
+$testingDate = array($year."-".$month."-".$day." ".$hour.":".$minute.":".$second.".".$frac.$frac2,
+                      $year."-".$month."-".$day." ".$hour.":".$minute.":".$second.".".$frac."0000",
+                      $year."-".$month."-".$day." ".$hour.":".$minute.":".$second.".000000",
+                      $year."-".$month."-".$day." ".$hour.":".$minute.":00.000000",
+                      $year."-".$month."-".$day." ".$hour.":".$minute.":".$second.".".$frac.$frac2." ".$tz_correction,
+                      $year."-".$month."-".$day." ".$hour.":".$minute.":".$second.".".$frac."0000 ".$tz_correction,
+                      $year."-".$month."-".$day." ".$hour.":".$minute.":".$second.".000000 ".$tz_correction,
+                      $year."-".$month."-".$day." ".$hour.":".$minute.":00.000000 ".$tz_correction,
+                      );
+ 
+// These formats are for the ODBC driver with types specified in sqlsrv_prepare()
+$date_formats = array($year."-".$month."-".$day
+                      );
+$time_formats = array($hour.":".$minute.":".$second,
+                      $hour.":".$minute.":".$second.".".$frac,
+                      $hour.":".$minute.":".$second.".".$frac.$frac2
+                      );
+
+// These formats are not accepted by either the ODBC driver or by PHP, but
+// can possibly be wrangled in sqlsrv_prepare() using strings instead of
+// the dedicated date and time types.
+$date_formats_nonODBC = array($year."/".$month."/".$day,
+                              $month."/".$day."/".$year,
+                              $month."-".$day."-".$year,
+                              $day."/".$month."/".$year,
+                              $day."-".$month."-".$year,
+                              $month_name." ".$day.", ".$year,
+                              $day."-".$month_name."-".$year,
+                              $day."-".$month_abbr."-".$year
+                              );
+$time_formats_nonODBC = array($hour12.":".$minute." ".$meridian,
+                              $hour12.":".$minute.":".$second." ".$meridian,
+                              $hour12.":".$minute.":".$second.".".$frac." ".$meridian,
+                              $hour12.":".$minute.":".$second.".".$frac.$frac2." ".$meridian,
+                              $hour.":".$minute
```

Comment:
Please enlighten me what these nonODBC formats are for.

Reply:
Some of them do work when passed in to sqlsrv_prepare as strings.